streams/src/main/java/org/apache/kafka/streams/StoreQueryParams.java

```diff
@@ -35,46 +34,42 @@
     private final String storeName;
     private final QueryableStoreType<T> queryableStoreType;
 
-    private StoreQueryParams(final String storeName, final QueryableStoreType<T>  queryableStoreType) {
+    private StoreQueryParams(final String storeName, final QueryableStoreType<T>  queryableStoreType, final Integer partition, final boolean staleStores) {
         this.storeName = storeName;
         this.queryableStoreType = queryableStoreType;
+        this.partition = partition;
+        this.staleStores = staleStores;
     }
 
     public static <T> StoreQueryParams<T> fromNameAndType(final String storeName,
                                                           final QueryableStoreType<T>  queryableStoreType) {
-        return new StoreQueryParams<T>(storeName, queryableStoreType);
+        return new StoreQueryParams<T>(storeName, queryableStoreType, null, false);
```

Comment:
Could we use a different sentinel than `null` for the `partition` field?. Does an `Optional` make sense?

Reply:
Actually we had a long discussion on keeping this null and adding it in Java doc that null means all partitions. Do you think this is important to start a discussion on the thread again? :)

Reply:
dont worry about it :) .. Happy to budge on stylistic things and leave it to the committers' call